docs/core/integration-quality-scale/rules/iqs017.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "IQS017 - Services raise errors when encountering failures"
+---
+
+IQS017 is as follows:
+
+Services raise errors when encountering failures
+
+## Reasoning
+
+Things can go wrong when a service is executed.
+When this happens, the integration should raise an error to indicate that something went wrong.
+These will be shown to the user in the UI, and can be used to help diagnose the issue.
+
+## Example implementation
+
+When the exception is caused by incorrect usage (for example incorrect input or referencing something that does not exist) should raise a `ServiceValidationError`.
+When the exception is caused by an error in the service itself (for example a network error or a bug in the service) should raise a `HomeAssistantError`.
+
+In this example, we show a function that is registered as a service in Home Assistant.
+If the input is incorrect (when the end date is before the start date), a `ServiceValidationError` is raised and if we can't reach the service, we raise a `HomeAssistantError`.
+
+```python
+async def async_set_schedule(call: ServiceCall) -> ServiceResponse:
+    """Set the schedule for a day."""
+    start_date = call.data[ATTR_START_DATE]
+    end_date = call.data[ATTR_END_DATE]
+    if end_date < start_date:
+        raise ServiceValidationError("End date must be after start date")
```

Comment:
I think we need to showcase best practices (and thus use translations from the get-go).

Reply:
I agree, but I haven't done that consistent. For example, in IQS001, the config flow, I am also not making an ideal config flow where we check a connection and check for already existing integrations